src/common/vector_operations/vector_cast.cpp

```diff
@@ -633,7 +633,7 @@ void FillEnum(Vector &source, Vector &result, idx_t count) {
 			result_mask.SetInvalid(i);
 			continue;
 		}
-		auto str = str_vec[source_data[src_idx]];
+		auto str = str_vec.GetValue(source_data[src_idx]).ToString();
```

Comment:
`GetValue`, `ToString`, etc are all very slow as this means multiple copies of the string are made. Could we turn this into a proper loop that uses the internal `string_t` type?